plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcMetadata.java

```diff
@@ -303,13 +313,63 @@ private JdbcTableHandle flushAttributesAsQuery(ConnectorSession session, JdbcTab
         handle = new JdbcTableHandle(
                 handle.getRelationHandle(),
                 handle.getConstraint(),
+                handle.getSortOrder(),
                 OptionalLong.of(limit),
                 handle.getColumns(),
                 handle.getNextSyntheticColumnId());
 
         return Optional.of(new LimitApplicationResult<>(handle, jdbcClient.isLimitGuaranteed(session)));
     }
 
+    @Override
+    public Optional<TopNApplicationResult<ConnectorTableHandle>> applyTopN(
+            ConnectorSession session,
+            ConnectorTableHandle table,
+            long topNCount,
+            List<SortItem> sortItems,
+            Map<String, ColumnHandle> assignments)
+    {
+        if (!isTopNPushdownEnabled(session)) {
+            return Optional.empty();
+        }
+
+        verify(!sortItems.isEmpty(), "sortItems are empty");
+        JdbcTableHandle handle = (JdbcTableHandle) table;
+
+        List<SortItem> sortOrder = sortItems.stream()
```

Comment:
The variables `List<SortItem> sortItems` and `List<SortItem> sortOrder` have different names (obviously), but the difference doesn't reflect the difference in meaning.
One uses assignment names to define sort items, and the other uses the column names.